openfasoc/generators/cryo-gen/flow/scripts/final_report.tcl

```diff
@@ -1,37 +1,59 @@
-if {![info exists standalone] || $standalone} {
-  # Read lef
-  read_lef $::env(TECH_LEF)
-  read_lef $::env(SC_LEF)
-  if {[info exist ::env(ADDITIONAL_LEFS)]} {
-    foreach lef $::env(ADDITIONAL_LEFS) {
-      read_lef $lef
-    }
-  }
-
-  # Read liberty files
-  foreach libFile $::env(LIB_FILES) {
-    read_liberty $libFile
-  }
-
-  # Read def and sdc
-  # Use -order_wires to build wire graph
-  # for antenna checker read_def -order_wires $::env(RESULTS_DIR)/6_1_fill.def
-  read_def $::env(RESULTS_DIR)/6_1_fill.def
-} else {
-  puts "Starting final report"
-}
+utl::set_metrics_stage "finish__{}"
+source $::env(SCRIPTS_DIR)/load.tcl
+load_design 6_1_fill.odb 6_1_fill.sdc "Starting final report"
 
+set_propagated_clock [all_clocks]
 
 # Delete routing obstructions for final DEF
-source scripts/deleteRoutingObstructions.tcl
+source $::env(SCRIPTS_DIR)/deleteRoutingObstructions.tcl
 deleteRoutingObstructions
 
+write_db $::env(RESULTS_DIR)/6_final.odb
 write_def $::env(RESULTS_DIR)/6_final.def
 write_verilog $::env(RESULTS_DIR)/6_final.v
 
-report_design_area
-report_power
+# Run extraction and STA
+if {[info exist ::env(RCX_RULES)]} {
+
+  # Set RC corner for RCX
+  # Set in config.mk
+  if {[info exist ::env(RCX_RC_CORNER)]} {
+    set rc_corner $::env(RCX_RC_CORNER)
+  }
+  # RCX section
+  define_process_corner -ext_model_index 0 X
+  extract_parasitics -ext_model_file $::env(RCX_RULES)
+  # Write Spef
+  write_spef $::env(RESULTS_DIR)/6_final.spef
+  file delete $::env(DESIGN_NAME).totCap
+  # Read Spef for OpenSTA
+  read_spef $::env(RESULTS_DIR)/6_final.spef
+  # Static IR drop analysis
+  if {[info exist ::env(PWR_NETS_VOLTAGES)]} {
+    dict for {pwrNetName pwrNetVoltage}  {*}$::env(PWR_NETS_VOLTAGES) {
+        set_pdnsim_net_voltage -net ${pwrNetName} -voltage ${pwrNetVoltage}
+        analyze_power_grid -net ${pwrNetName}
+    }
+  } else {
+    puts "IR drop analysis for power nets is skipped because PWR_NETS_VOLTAGES is undefined"
+  }
+  if {[info exist ::env(GND_NETS_VOLTAGES)]} {
+    dict for {gndNetName gndNetVoltage}  {*}$::env(GND_NETS_VOLTAGES) {
+        set_pdnsim_net_voltage -net ${gndNetName} -voltage ${gndNetVoltage}
+        analyze_power_grid -net ${gndNetName}
+    }
+  } else {
+    puts "IR drop analysis for ground nets is skipped because GND_NETS_VOLTAGES is undefined"
+  }
+
+} else {
+  puts "OpenRCX is not enabled for this platform."
+}
+
+source $::env(SCRIPTS_DIR)/report_metrics.tcl
+report_metrics "finish"
 
-if {![info exists standalone] || $standalone} {
-  exit
+# Save a final image if openroad is compiled with the gui
```

Comment:
@Jerry-lqx 
I had error when running on my machine (may be relate to OR gui) and I traced it to this line. The error ends program execution without LVS/DRC. I do not think this is very important and I recommend removing... thoughts?